### DISCUSSION TITLE
Date separator is showing two times on the same day

### DISCUSSION THREAD

**Discussion by ramanareddy463** [GetStream/stream-chat-android#4668] [General]
The date separator is showing two times on the same day.
I am using XML-based Chat UI components (io.getstream.chat.android.ui.message.list.MessageListView)

Steps 1. Send multiple messages at 11:30 AM.
           2. Send again messages at 4:00 PM.
           
 If following the above steps can see the two date separators.  
 ========================================
 I used below method, but not call the method
 
 `messageListViewModel.setDateSeparatorHandler { previousMessage, message -> 
 
 }`

**Comment by MarinTolic** [comment]
Hi @ramanareddy463 ,

This is not a bug, rather the time period between messages required for the date separator to show is set to 4 hours by default.
If you wish to change the time period, you can do so by providing your own date separator handlers.

To override the date separator handler when the list is in  "normal" mode call `MessageListViewModel.setDateSeparatorHandler()`, and to override it when the list is in thread mode call `MessageListViewModel.setThreadDateSeparatorHandler()`.

Kind regards,
Marin.

**Comment by ramanareddy463** [comment]
Thanks for reply @MarinTolic 
As mentioned in the issue, I already override date separator handlers.

```
 val calendar = Calendar.getInstance()
 messageListViewModel.setDateSeparatorHandler {  previousMessage, message ->
 
            if (previousMessage == null) {
                true
            } else {
                shouldShowDateSeparator(calendar, previousMessage, message)
            }
            
        }
        messageListViewModel.setThreadDateSeparatorHandler {  previousMessage, message ->
        
            if (previousMessage == null) {
                false
            } else {
                shouldShowDateSeparator(calendar, previousMessage, message)
            }
            
        }
        
        
  private fun shouldShowDateSeparator(calendar: Calendar, previousMessage: Message, message: Message): Boolean {
        val (previousYear, previousDayOfYear) = calendar.run {
            time = previousMessage.getCreatedAtOrThrow()
            get(Calendar.YEAR) to get(Calendar.DAY_OF_YEAR)
        }
        val (year, dayOfYear) = calendar.run {
            time = message.getCreatedAtOrThrow()
            get(Calendar.YEAR) to get(Calendar.DAY_OF_YEAR)
        }
        return previousYear != year || previousDayOfYear != dayOfYear
    }
 
```
These overrise handlers not calling/responding..
Can you please guide me, am i missing anything.

Regards,
Ramana.

**Reply by MarinTolic** [reply]
Hey @ramanareddy463 ,

This was indeed a bug you found.
It appears we haven't been updating the handlers properly after they were set.

The fix is contained in PR https://github.com/GetStream/stream-chat-android/pull/4681 and will be available in the next release.
